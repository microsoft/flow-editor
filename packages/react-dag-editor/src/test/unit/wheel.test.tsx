import { cleanup, fireEvent, render } from "@testing-library/react";
import * as React from "react";

import { act } from "react-dom/test-utils";
import { GraphCanvasEvent } from "../../index";
import { TestComponent } from "../TestComponent";

beforeAll(() => {
  jest.useFakeTimers();
});

beforeEach(() => {
  jest.spyOn(window, "requestAnimationFrame").mockImplementation((cb) => {
    cb(0);
    return 0;
  });
});

afterEach(cleanup);

describe("Wheel", () => {
  it("should not work with wheel event without focus", () => {
    const onEvent = jest.fn();
    const { container } = render(<TestComponent graphProps={{ onEvent }} />);
    const svg = container.querySelector("svg")!;
    expect(svg).toBeTruthy();

    document.body.focus();

    act(() => {
      fireEvent.wheel(svg, {
        deltaY: 200,
      });
      jest.runAllTimers();
    });

    expect(onEvent).not.toHaveBeenCalledWith(
      expect.objectContaining({
        type: GraphCanvasEvent.MouseWheelScroll,
      })
    );
  });
});
