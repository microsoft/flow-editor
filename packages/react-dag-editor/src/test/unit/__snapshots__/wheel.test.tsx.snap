// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wheel should not work with wheel event without focus: events 1`] = `
Array [
  "[Canvas]ViewportResize",
  "[Canvas]VirtualizationRecalculated",
]
`;

exports[`Wheel should work with wheel event after focus graph: events 1`] = `
Array [
  "[Canvas]ViewportResize",
  "[Canvas]VirtualizationRecalculated",
  "[Canvas]Focus",
  "[Canvas]MouseWheelScroll",
  "[Canvas]VirtualizationRecalculated",
]
`;
